client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/steps/use-steps.tsx

```diff
@@ -0,0 +1,146 @@
+import { ExternalLink } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import type { Task, Expandable } from '@automattic/launchpad';
+
+const removeDuplicatedSlashes = ( url: string ) => url.replace( /(https?:\/\/)|(\/)+/g, '$1$2' );
+
+const getPluginInstallationPage = ( fromUrl: string ) => {
+	if ( fromUrl !== '' ) {
+		return removeDuplicatedSlashes(
+			`${ fromUrl }/wp-admin/plugin-install.php?s=%2522migrate%2520guru%2522&tab=search&type=term`
+		);
+	}
+
+	return 'https://wordpress.org/plugins/migrate-guru/';
+};
+
+const getMigrateGuruPageURL = ( siteURL: string ) =>
+	removeDuplicatedSlashes( `${ siteURL }/wp-admin/admin.php?page=migrateguru` );
+
+const recordInstructionsLinkClick = ( linkname: string ) => {
+	recordTracksEvent( 'calypso_site_migration_instructions_link_click', {
+		linkname,
+	} );
+};
+
+interface Options {
+	fromUrl: string;
+}
+
+interface Step {
+	task: Task;
+	expandable?: Expandable;
+}
+
+type Steps = Step[];
+
+export const useSteps = ( { fromUrl }: Options ): Steps => {
+	const translate = useTranslate();
+
+	const steps = [
+		{
+			title: translate( 'Install the Migrate Guru plugin' ),
+			content: (
+				<p>
+					{ translate(
+						"First you'll need to install and activate the {{a}}Migrate Guru plugin{{/a}} on the site you want to migrate. Click {{strong}}Next{{/strong}} when you're ready.",
+						{
+							components: {
+								strong: <strong />,
+								a: (
+									<ExternalLink
+										href={ getPluginInstallationPage( fromUrl ) }
+										target="_blank"
+										onClick={ () => recordInstructionsLinkClick( 'install-plugin' ) }
+									/>
+								),
+							},
+						}
+					) }
+				</p>
+			),
+		},
+		{
+			title: translate( 'Get your site ready' ),
+			content: (
+				<>
+					<p>
+						{ translate(
+							'Head to the {{a}}Migrate Guru plugin screen on your source site{{/a}}, enter your email address, and click {{strong}}%(migrateLabel)s{{/strong}}.',
+							{
+								components: {
+									strong: <strong />,
+									a: fromUrl ? (
+										<ExternalLink
+											href={ getMigrateGuruPageURL( fromUrl ) }
+											target="_blank"
+											onClick={ () => recordInstructionsLinkClick( 'go-to-plugin-page' ) }
+										/>
+									) : (
+										<strong />
+									),
+								},
+								args: { migrateLabel: 'Migrate' },
+							}
+						) }
+					</p>
+					<p>
+						<strong>{ translate( 'Then, pick WordPress.com as your destination host.' ) }</strong>
```

Comment:
Curious as to why this text is emphasized? It seems like we're using that for just the step titles and the button text.

Reply:
Good point. You're right that it doesn't follow our other patterns for bold. I just assumed to leave it as is. I think we can remove the bold formatting for it.

Reply:
It came from the first versions of the design. Maybe @fditrapani could help us to clarify. :)

Reply:
Oh, sorry! My interface didn't update and I didn't see the Filippo's answer before adding mine. haha
I'm removing the bold from this point then.

Reply:
Done https://github.com/Automattic/wp-calypso/pull/92070/commits/8570ecc8f4e6d8c581c9676cc71c455657f6c5e8